python/ml_wrappers/model/image_model_wrapper.py

```diff
@@ -249,53 +248,93 @@ def predict_proba(self, dataset: pd.DataFrame) -> np.ndarray:
-            detections = self._model(ToTensor()(image).to(self._device).unsqueeze(0))
-
-            prediction = detections[0]
-
-            keep = torchvision.ops.nms(
-                prediction["boxes"],
-                prediction["scores"],
-                iou_threshold,
-            )
-
-            prediction["boxes"] = prediction["boxes"][keep]
-            prediction["scores"] = prediction["scores"][keep]
-            prediction["labels"] = prediction["labels"][keep]
-
-            image_predictions = torch.cat((prediction["labels"].unsqueeze(1),
-                                           prediction["boxes"],
-                                           prediction["scores"].unsqueeze(1)), dim=1)
-
-            predictions.append(image_predictions.detach().cpu().numpy().tolist())
-
-        return predictions
+        detections = []
+        for image in x:
+            if type(image) == torch.Tensor:
+                raw_detections = self._model(image.to(self._device).unsqueeze(0))
+            else:
+                raw_detections = self._model(T.ToTensor()(image)
+                                             .to(self._device).unsqueeze(0))
+
+            def apply_nms(orig_prediction: dict):
+                """Perform nms on the predictions based on their IoU.
+
+                :param orig_prediction: Original model prediction
+                :type orig_prediction: dict
+                :param iou_thresh: iou_threshold for nms
+                :type iou_thresh: float
+                :return: Model prediction after nms is applied
+                :rtype: dict
+                """
+                keep = torchvision.ops.nms(orig_prediction['boxes'],
+                                           orig_prediction['scores'], iou_thresh)
+
+                nms_prediction = orig_prediction
+                nms_prediction['boxes'] = nms_prediction['boxes'][keep]
+                nms_prediction['scores'] = nms_prediction['scores'][keep]
+                nms_prediction['labels'] = nms_prediction['labels'][keep]
+                return nms_prediction
+
+            def filter_score(orig_prediction: dict):
+                """Filter out predictions with confidence scores < score_thresh.
+
+                :param orig_prediction: Original model prediction
+                :type orig_prediction: dict
+                :param score_thresh: Score threshold to filter by
+                :type score_thresh: float
+                :return: Model predictions filtered out by score_thresh
+                :rtype: dict
+                """
+                keep = orig_prediction['scores'] > score_thresh
+
+                filter_prediction = orig_prediction
+                filter_prediction['boxes'] = filter_prediction['boxes'][keep]
+                filter_prediction['scores'] = filter_prediction['scores'][keep]
+                filter_prediction['labels'] = filter_prediction['labels'][keep]
+                return filter_prediction
```

Comment:
Consider pushing `apply_nms` and `filter_score` outside `predict`'s definition and invoke `self.{function}` instead.